modules/gapi/include/opencv2/gapi/gasync_context.hpp

```diff
@@ -0,0 +1,37 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+#ifndef OPENCV_GAPI_GASYNC_CONTEXT_HPP
+#define OPENCV_GAPI_GASYNC_CONTEXT_HPP
+
+#if !defined(GAPI_STANDALONE)
+#  include <opencv2/core/cvdef.h>
+#else   // Without OpenCV
+#  include <opencv2/gapi/own/cvdefs.hpp>
+#endif // !defined(GAPI_STANDALONE)
+
+#include <opencv2/gapi/own/exports.hpp>
+
+namespace cv {
+namespace gapi{
+namespace wip {
+
+class GAPI_EXPORTS GAsyncContext{
+    std::atomic<bool> cancelation_requested = {false};
+public:
+    bool cancel();
+    bool isCanceled() const;
+};
```

Comment:
Documentation, please. What does `bool` return value mean for `cancel()`?

We also discussed that the context destructor should do cancellation (see mail notes). Is it implemented?

Reply:
Added a comment on returning `bool`. 

As for "context destructor should do cancellation" - no it is not implemented and most likely will not be implemented. (Well, long story short  - it is a __bad__ idea which does not allow to achieve desired behaviour - will write a separate e-mail)

Reply:
okay, let's discuss what's remaining

Reply:
It is not a doxygen-capable documentation, unfortunately.

Please open an issue to document this API part properly (with correct doxygen documentation and an API sample in it - see our other public headers)